### PR TITLE
[HDRP] Light anchor fixes

### PR DESCRIPTION
**Pull request by alelievr**
### Purpose of this PR

- Fixed algorithm so that a distance of 0 doesn't screw the rotation
- Fixed Light anchor fields not working in prefabs (except orbit, elevation, and yaw which are not serialized in this component)
- Remove the dependency on the light component
- Added an anchor position override transform component to have a physical target in the scene.
- Fixed the anchor rotation being reset when the object is copy/pasted
- Light anchor modifier tool now works correctly with rotation pivot mode

https://fogbugz.unity3d.com/f/cases/1345509/

Note that if the anchor position override is not null, the light anchor transform will be updated each time the target transform updates.

---
### Testing status

tested that the light anchor distance works correctly + `distance` and `Up direction` handle correctly prefabs.

https://user-images.githubusercontent.com/6877923/127169303-21f68267-98d7-4d26-b0f0-c5393daba179.mp4

https://user-images.githubusercontent.com/6877923/127310339-b1f693df-089a-46be-a5f9-98b730e4efe3.mp4

image

Note that there is an issue with Light Anchor prefabs when the distance is reverted, the transform of the gameobject is not updated using the reverted distance value (I'm not sure if we can fix that looking at how the component works currently)

The light angle is now adjusted with the camera position instead of rotation to correct the perspective

https://user-images.githubusercontent.com/6877923/128217714-32bf78d9-8eaa-424e-8d63-8a1de0f636ff.mp4

https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/hd%252Ffix%252Flight-anchor/.yamato%252Fall-hdrp.yml%2523PR_HDRP_trunk/8484994/job

### CODE REVIEW

File: com.unity.render-pipelines.core/Editor/Lighting/LightAnchorEditor.cs [hunk 1]
```diff
@@ -266,7 +268,7 @@ public override void OnInspectorGUI()
                 {
                     Undo.RecordObjects(new UnityEngine.Object[] { target, manipulator.transform }, "Light Anchor Change");
 
-                    manipulator.frameSpace = m_FrameSpace;
+                    manipulator.frameSpace = (LightAnchor.UpDirection)m_FrameSpaceProperty.enumValueIndex;
```
Comment (RSlysz) [inline com.unity.render-pipelines.core/Editor/Lighting/LightAnchorEditor.cs:271]: Check HDEditorUtils.GetEnumValue for details
Never use enumValueIndex to convert to type!
```suggestion
                    manipulator.frameSpace = (LightAnchor.UpDirection)m_FrameSpaceProperty.intValue;
```

**Review by RSlysz — CHANGES_REQUESTED** [review]

File: com.unity.render-pipelines.core/Runtime/Lights/LightAnchor.cs [hunk 2]
```diff
@@ -19,7 +18,7 @@ public class LightAnchor : MonoBehaviour
         const float k_ArcRadius = 5;
         const float k_AxisLength = 10;
 
-        [SerializeField]
+        [SerializeField, Min(0)]
```
[the remark this answers is not in the document]
Comment (alelievr) [reply com.unity.render-pipelines.core/Runtime/Lights/LightAnchor.cs:21]: No, we don't want the epsilon value to leak on the UI side

[the remark this answers is not in the document]
Comment (RSlysz) [reply com.unity.render-pipelines.core/Runtime/Lights/LightAnchor.cs:21]: So in Inspector you can put 0 but it is not supported when changing by scripts??
I insist. We must uniformize. If epsilon is not good, then 0 for both.

**Review by RSlysz — APPROVED** [review]

**Review by sebastienlagarde — APPROVED** [review]

**Review by skhiat — APPROVED** [review]

**Comment by iM0ve** [comment]
**Test status: In Progress**

**Concerns:**
1) The new "light look at" property is great, but the Pivot is used as the anchor point instead of Centre of the object. I assume in some cases this could be an issue? For instance in the video I wanted to focus the light on Adams head, but the Pivot position was at his feet so the light ended up looking towards the ground. At the end of the vid i show where is the centre vs pivot location


https://user-images.githubusercontent.com/36502659/131343620-f61b90f5-ee65-467a-9f20-ba14241ac940.mov

**Comment by pierre-unity** [comment]
I would suggest we also add an new "Position offset" under "Anchro Position Override" (indented), for all cases where the pivot of the "Anchor Position Override" is not exactly where it should logically be.

For instance, sometimes that pivot of a character is at the heal of the character, but the light should rather orbit the head. So with a Y-offset of roughly 1.6m, the light will be orbiting the right bit of the character.

**Comment by pierre-unity** [comment]
Another issue, related to very high distances: it creates a "pivot drift", which does not happen with a custom anchor position override btw.

https://user-images.githubusercontent.com/39901544/131544869-da007ebe-45f4-46c7-96c2-9dad421f983e.mp4

**Comment by pierre-unity** [comment]
Another potential improvement:

Can we get the list of all the bones from a gameobject, and pick it from a list? basically, if I pick an "Anchor Position Override" that has bones, we then display a drop down list of all bones attached to it and let the user pick a given bone transform.

See here how to get the list:
https://forum.unity.com/threads/solved-how-can-i-collect-bone-names-and-transforms.35810/#post-232041

**Comment by alelievr** [comment]
So we'll move forward and merge the PR as it is now and do further improvement in a second PR (This PR is already too old :) )

**Review by iM0ve — APPROVED** [review]

**Pull request merged**